### ISSUE TITLE
add a repl

### ISSUE THREAD

**Issue by fneddy** [gtk-ml/gtk-ml#13]
would be nice to have a repl commandline for gtk-ml.
steps that we should consider: 

- [ ] basic standalone usage 
- [ ] embedded in an application
- [ ] attach to running application

**Comment by walterpie** [comment]
Absolutely! Feel free to start implementing this. However, I started implementing gtk-ml as a library-only program, where applications dynamically link to it, so make sure the repl is usable as a library feature.

**Comment by walterpie** [comment]
:wink:

**Comment by walterpie** [comment]
We should also probably use GNU readline for that, because it's pretty good
